drop-in/src/test/java/com/adyen/checkout/dropin/ui/viewmodel/PaymentMethodsListViewModelTest.kt

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by atef on 27/10/2022.
+ */
+
+package com.adyen.checkout.dropin.ui.viewmodel
+
+import android.app.Application
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import app.cash.turbine.test
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.model.paymentmethods.StoredPaymentMethod
+import com.adyen.checkout.components.model.payments.Amount
+import com.adyen.checkout.dropin.DropInConfiguration
+import com.adyen.checkout.dropin.ui.paymentmethods.GiftCardPaymentMethodModel
+import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodHeader
+import com.adyen.checkout.dropin.ui.paymentmethods.PaymentMethodModel
+import com.adyen.checkout.dropin.ui.paymentmethods.StoredPaymentMethodModel
+import com.adyen.checkout.test.TestDispatcherExtension
+import com.adyen.checkout.util.ConfigurationProvider
+import com.adyen.checkout.util.DataProvider
+import com.adyen.checkout.util.Helpers.mapToPaymentMethodModelList
+import com.adyen.checkout.util.Helpers.mapToStoredPaymentMethodsModelList
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.junit.jupiter.MockitoExtension
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(MockitoExtension::class, TestDispatcherExtension::class)
+internal class PaymentMethodsListViewModelTest {
```

Comment:
Good tests overall 👍 Can we also test some more scenarios, especially the ones that caused us to start this fix? Like passing only `paymentMethods` or only `storedPaymentMethods` or a non null `order`, etc?

Reply:
they are already add in the inner class `PaymentMethodFLow`

Reply:
Oops sorry my local version was out of date 😅

Reply:
Then one more test with a non null `order` would make this 💯 🙏